### PR TITLE
add topic remapping option to rosbag2 play

### PR DESCRIPTION
**Pull request by ketatam**
With the option --remap or -m it's now possible to do topic remapping in rosbag2 play.
This is done by passing the remapping options to the play node.

Closes https://github.com/ros2/rosbag2/issues/326.

### CODE REVIEW

File: rosbag2_transport/include/rosbag2_transport/rosbag2_transport.hpp [hunk 1]
```diff
@@ -74,7 +74,9 @@ class Rosbag2Transport
    * \param play_options Options regarding the playback (e.g. queue size)
    */
   ROSBAG2_TRANSPORT_PUBLIC
-  void play(const StorageOptions & storage_options, const PlayOptions & play_options);
+  void play(
+    const StorageOptions & storage_options, const PlayOptions & play_options,
+    const std::vector<std::string> & topic_remapping_options = {});
```
Comment (emersonknapp) [inline rosbag2_transport/include/rosbag2_transport/rosbag2_transport.hpp:79]: Do you think it would be more appropriate to add `topic_name_map` to `PlayOptions`/`RecordOptions` rather than adding new arguments to the API?

File: ros2bag/ros2bag/verb/play.py [hunk 2]
```diff
@@ -52,6 +52,10 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             '-l', '--loop', action='store_true',
             help='enables loop playback when playing a bagfile: it starts back at the beginning '
                  'on reaching the end and plays indefinitely.')
+        parser.add_argument(
```
Comment (emersonknapp) [inline ros2bag/ros2bag/verb/play.py:55]: should we call this `--ros-args` (and call out that it enables topic name remapping in the `help` string), since you can pass any `ros-args` here and it'll work?

Comment (Martin-Idel) [reply ros2bag/ros2bag/verb/play.py:55]: Yes, we were wondering that. My reasoning not to go that route was the following:
- `--ros-args` was introduced in particular to avoid name clashes with user defined flags on user ros2 nodes, in other words to separate flags defined by rclcpp or rclpy, etc. from flags defined on nodes deriving from those
- Rosbag isn't the classical node and I don't see a scenario where you'd want to provide a custom rosbag node, so separating "namespaces" of flags is not a big deal here
- It could be interesting to expose more flags from rosbag's node, but I wonder whether it is a good idea to expose all of them. Therefore, I'd say: keep things hidden, so we'd only expose `--ros-args --remap` for now. I also wondered whether at that point it isn't confusing if the notation is the same for a general node but not all options you can generally pass to `--ros-args` are available.
What do you think about this?

Comment (ketatam) [reply ros2bag/ros2bag/verb/play.py:55]: I have added added a new topic remapping options to the play options. Please let me know if there are still some other problems

File: rosbag2_transport/include/rosbag2_transport/rosbag2_transport.hpp [hunk 3]
```diff
@@ -74,7 +74,8 @@ class Rosbag2Transport
    * \param play_options Options regarding the playback (e.g. queue size)
    */
   ROSBAG2_TRANSPORT_PUBLIC
-  void play(const StorageOptions & storage_options, const PlayOptions & play_options);
+  void play(
+    const StorageOptions & storage_options, const PlayOptions & play_options);
```
Comment (emersonknapp) [inline rosbag2_transport/include/rosbag2_transport/rosbag2_transport.hpp:78]: Remove unrelated change
```suggestion
  void play(const StorageOptions & storage_options, const PlayOptions & play_options);
```

Comment (ketatam) [reply rosbag2_transport/include/rosbag2_transport/rosbag2_transport.hpp]: done!

**Comment by Martin-Idel** [comment]
CI (gist: https://gist.githubusercontent.com/Martin-Idel/f92c7fb9c27def6447c87d222532d75c/raw/c42febac209f511b0b28d03628e2ccd2b6571e42/ros2_rosbag_topic_remapping.repos):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10515/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5982/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8567/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10389/)

**Comment by ketatam** [comment]
@emersonknapp  the CI is unstable but I think it's not because of my code. So can you please check this and if I do have to change somthing just let me know. Thanks!

**Comment by emersonknapp** [comment]
I took a look through the builds and yes, I don't see anything here that's specifically related to this PR.

The Action CI is the continuing https://github.com/ros2/rosbag2/issues/381
OSX and Windows build warnings are all in Fast-RTPS
The consistent test failure across platforms seems to be an issue with Connext type support, I don't want to block this change on any changes happening upstream there.

We'll need to keep an eye on these issues leading up to the release candidate package, but for now I think we're good to merge.
